waku/v2/protocol/filter/filter_health_check.go

```diff
@@ -0,0 +1,37 @@
+package filter
+
+import (
+	"context"
+	"time"
+)
+
+func (wf *WakuFilterLightNode) PingPeers() {
+	//Send a ping to all the peers and report their status to corresponding subscriptions
+	// Alive or not or set state of subcription??
+	for _, peer := range wf.subscriptions.GetSubscribedPeers() {
+		err := wf.Ping(context.TODO(), peer)
+		if err != nil {
+			subscriptions := wf.subscriptions.GetAllSubscriptionsForPeer(peer)
+			for _, subscription := range subscriptions {
+				//Indicating that subscription is closing
+				//This feels like a hack, but taking this approach for now so as to avoid refactoring.
+				subscription.Closing <- true
```

Comment:
That is to ensure each subscription that is subscribed to that peer can be resubscribed. 
e.g: S1 and S2 are subscribed to P1. Now if ping fails for P1, we need to resubscribe both S1 and S2 again to another peer. 

I am not sure if closing the peer connection is the right thing to do though. Need to think more, because what if that peer is also a store node and just not able to process filter pings alone. (Maybe due to rate-limit or something else)
Can we take a decision just based on filter ping?